sqlglot/expressions.py

```diff
@@ -306,7 +308,7 @@ def find(self, *expression_types, bfs=True):
         """
         return next(self.find_all(*expression_types, bfs=bfs), None)
 
-    def find_all(self, *expression_types, bfs=True):
+    def find_all(self, *expression_types: t.Type[E], bfs=True) -> t.Generator[E, None, None]:
```

Comment:
t.Iterable[E]

Reply:
👍 i can make the return type `t.Iterator[E]` ... `t.Iterable[E]` will fail to type check because an iterator doesn't support `next`, which is used above on line 309